File: ClientApp/src/app/interfaces/security/Response/userResponse.ts
import { ByBaseResponse } from "../../byBaseResponse";
import { UserRoleResponse } from "./userRoleResponse";

export interface UserResponse extends ByBaseResponse {
	id: string;
	userName: string;
	fullName: string;
	email: string;
	birthDate: Date;
	promotionalMails: boolean;
	userRoles: UserRoleResponse[];
}

File: Contracts/Model/Security/Response/UserResponse.cs
namespace Contracts.Model.Security.Response;

public class UserResponse : ByBaseResponse
{
	public Guid Id { get; set; }
	public string UserName { get; set; }
	public string Email { get; set; }
	public DateOnly BirthDate { get; set; }
	public bool PromotionalMails { get; set; }
	public IEnumerable<UserRoleResponse> UserRoles { get; set; }
}
